src/backend/catalog/heap.c

```diff
@@ -385,14 +386,31 @@ heap_create(const char *relname,
 	 */
 	if (create_storage)
 	{
+		RelFileLocator prev_rlocator = rel->rd_locator;
+		RelFileLocator new_rlocator = rel->rd_locator;
+
+		if (old_rlocator != NULL)
+		{
+			prev_rlocator = *old_rlocator;
+
+			/*
+			 * table_relation_set_new_filelocator() takes old_rlocator from
+			 * rel->rd_locator
+			 */
+			rel->rd_locator = prev_rlocator;
```

Comment:
Maybe it would be better to do this similarly to the OldOid in the heap create function, but this is a good workaround for now so that we can get the release progressing.